source/wp-content/themes/wporg-news-2021/block-templates/index.html

```diff
@@ -1,12 +1,6 @@
-<!-- wp:wporg/global-header /-->
-
-<!-- wp:group {"tagName":"header","className":"header-archive header-archive-all-posts","layout":{"inherit":true}} -->
-<header class="wp-block-group header-archive header-archive-all-posts">
-	<!-- wp:heading {"level":1,"align":"wide"} -->
-	<h1 class="alignwide">All Posts</h1>
-	<!-- /wp:heading -->
-</header>
-<!-- /wp:group -->
+<!-- wp:template-part {"slug":"header","align":"full","className":"site-header-container"} /-->
+<!-- This is a hack to ensure content isn't covered by the fixed header -->
+<!-- wp:template-part {"slug":"header","align":"full","className":"site-header-offset"} /-->
```

Comment:
@iandunn If there's a better way to do this, I'm certainly open to it. But if we do go with this approach, one thing that will need to be changed is how the template file for the global header is included. Right now it uses `require_once`, but that means that its markup doesn't get output a second time within `site-header-offset` because it's already been loaded.

Reply:
todo test to understant why doing this

--

clarify docs why it's needed
open G issue for bug or whatever, link to it in comment

Reply:
This is a creative approach, but it seems safe to assume that the local nav will always have a known/stable height. If that's the case, then this seems to solve the problem in a more conventional and obvious way:

```css
.site-content-container {
   // Fixed position elements are removed from the normal document flow, so we have to account for it in order to avoid content being hidden under it.
   margin-top: var( --wp-site-header-height );
}
```

Is that not the case?

Reply:
The local nav should mostly have a stable height (expanded category list notwithstanding, but that's intended to overlap the content anyway). However, on narrow screens, the height changes when the breadcrumb and the categories toggle stack. Furthermore, the global header (also fixed) can also have a variable height at different viewport widths, if I'm not mistaken.

Reply:
Ah, you're right. I don't see any good CSS-only approaches for the dynamic height problem ☹️ 

I wouldn't be surprised if we could hardcode an educated guess that'd work in practical use cases, but the duplicate-content approach WFM too.

In that case, it might help to link to something like https://stackoverflow.com/questions/26163818/css-fixed-header-of-unknown-height/26164057 in the comments, just to clarify the problem and the tradeoffs with various solutions